plugin/evm/orderbook/contract_events_processor.go

```diff
@@ -471,11 +465,22 @@ func (cep *ContractEventsProcessor) ProcessTraderEvents(events []*types.Log, blo
 					continue
 				}
 				eventName = "OrderPlaced"
+				order := IOCOrder{}
+				order.DecodeFromRawOrder(args["order"])
+				args["order"] = order.Map()
 			}
 		default:
 			continue
 		}
+		if len(event.Topics) > 1 {
+			trader = getAddressFromTopicHash(event.Topics[1])
+		}
+		if len(event.Topics) > 2 {
+			orderId = event.Topics[2]
+		}
+
```

Comment:
this is still slightly risky, if we add any event in orderbook contracts that has different first fields this will panic